src/tmrc/logging_config.py

```diff
@@ -0,0 +1,128 @@
+import logging
+import socket
+import sys
+
+LOGGER_NAME = "tmrc"
+LOGGING_LEVEL = logging.DEBUG
+LOGGING_FORMAT = "%(asctime)s %(hostname)s(%(local_rank)d) %(levelname)s -> %(message)s"
+LOGGING_STREAM = sys.stdout
+LOGGING_STREAM_LEVEL = logging.INFO
+LOGGING_FILE_LEVEL = logging.DEBUG
+
+
+# Set up rank 0 filter
+class RankBasedFilter(logging.Filter):
+    def filter(self, record):
+        """
+        Filter to only allow logs from rank 0.
+        """
+        if record.levelno > logging.INFO:
+            return True
+        if getattr(record, "all_rank", None):
+            return True
+        return record.global_rank == 0
+
+
+class LoggerManager:
+    """
+    A singleton class to manage the logger instance.
+    This class ensures that only one logger instance is created and used
+    throughout the application.
+    """
+
+    _instance: "LoggerManager" = None
+
+    def __new__(cls):
```

Comment:
Is this really necessary? The logger is already implemented as a singleton under the hood.

For example see this docs: https://docs.python.org/3/howto/logging-cookbook.html?utm_source=#creating-a-lot-of-loggers

During script execution, the following snippet will always return the same logger instance:
```
import logging
logger = logging.getLogger("my_logger")
```
So there's no need to manually enforce singleton behavior. `getLogger()` already does it for you.